language/move-prover/boogie-backend/src/prelude/native.bpl

```diff
@@ -425,10 +422,11 @@ function {:inline} {{impl.fun_spec_has_key}}{{S}}(t: ({{Self}}), k: {{K}}): bool
 
 {%- if impl.fun_spec_set != "" %}
 function {:inline} {{impl.fun_spec_set}}{{S}}(t: {{Self}}, k: {{K}}, v: {{V}}): {{Self}} {
-    if (ContainsTable(t, {{ENC}}(k))) then
-        UpdateTable(t, {{ENC}}(k), v)
+    (var enc_k := {{ENC}}(k);
```

Comment:
I am just curious if this change was also necessary to resolve the inconsistency issue.